File: 백준/Silver/14916. 거스름돈/거스름돈.py
import sys
N = int(sys.stdin.readline())    

result = 0

while N % 5 != 0 :
    if(N < 0) :
        break
    result += 1
    N -= 2

result += N // 5

if N % 5 == 0 :
    print(result)
else :
    print(-1)

File: 백준/Bronze/5585. 거스름돈/거스름돈.java
import java.io.*;
import java.util.*;

public class Main {
	static int[] coin = {500,100,50,10,5,1};
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
        int N = 1000 - Integer.parseInt(br.readLine());
		int result = 0;
		
		for (int i : coin) {
			result += N/i;
			N%=i;
		}
		System.out.println(result);
		br.close();
	}
}